packages/core/src/hooks/useQueryParamState.ts

```diff
@@ -64,6 +65,14 @@ export function useQueryParamState<T>(
     extractState(location.search, stateName),
   );
 
+  useEffect(() => {
+    const newState = extractState(location.search, stateName);
+
+    setQueryParamState(oldState =>
+      isEqual(newState, oldState) ? oldState : newState,
+    );
+  }, [location, stateName]);
+
```

Comment:
This change is required to make sure that the sidebar search also work when the user is already on the search route. @benjdlambert mentioned this before, but now we have a place were I can reproduce this issue.